libcxx/src/filesystem/filesystem_clock.cpp

```diff
@@ -29,7 +29,8 @@
 #  include <sys/time.h> // for gettimeofday and timeval
 #endif
 
-#if defined(__APPLE__) || defined(__gnu_hurd__) || (defined(_POSIX_TIMERS) && _POSIX_TIMERS > 0)
+#if defined(__APPLE__) || defined(__gnu_hurd__) || defined(_LIBCPP_TARGETING_GPU) ||                                   \
```

Comment:
Seems like we should file a bug against whoever provides those headers for them to add it, then? BTW, I'm missing some context but on these platforms, who/what is providing these platform headers? Do you compile using some kind of SDK?

Reply:
Me, I provide them through https://libc.llvm.org/gpu/index.html. I don't include `unistd.h` because it's POSIX and pretty much nothing in there has any utility being run on a GPU. I provide `clock_gettime` which works around this because it can actually be implemented (At least for monotonic) and it's in `time.h`, which is header we have anyway.